interfaces/dbus/backend.go

```diff
@@ -132,6 +156,68 @@ func addContent(securityTag string, executableSnippets [][]byte, content map[str
 	}
 }
 
+func (b *Backend) setupBusservices(snapInfo *snap.Info, snippets map[string][][]byte) error {
+	snapName := snapInfo.Name()
+
+	content := map[string]*osutil.FileState{}
+	for _, appInfo := range snapInfo.Apps {
+		securityTag := appInfo.SecurityTag()
+
+		// FIXME: layer violation, we really should do that in the
+		//        dbus interface but without PR #2613 its really tricky
+		for _, slot := range appInfo.Slots {
+			if slot.Interface != "dbus" {
+				continue
+			}
+			bus, ok := slot.Attrs["bus"].(string)
+			if !ok {
+				continue
+			}
+			if bus != "session" {
+				continue
+			}
+			name, ok := slot.Attrs["name"].(string)
+			if !ok {
+				continue
+			}
+			isService := slot.Attrs["service"].(bool)
```

Comment:
I added checks that its really bool and also added a comment why we need this. Should I do more (I'm not sure if that addresses the  cross-link comment).